src/main/kotlin/blackjack/BlackjackDealer.kt

```diff
@@ -0,0 +1,19 @@
+package blackjack
+
+class BlackjackDealer(private var deck: CardDeck = CardDeck.createDeck()) {
+    fun startTheGame(players: List<Player>) {
+        players.forEach(::drawTwice)
+    }
+
+    private fun drawTwice(player: Player) {
+        repeat(2) { player.addCardToHand(this.deck.draw()) }
+    }
+
+    fun shuffleCardDeck() {
+        this.deck = this.deck.shuffle()
+    }
```

Comment:
this는 주로 멤버변수 혹은 지역변수와 파라미터명이 동일할 때 사용하는 것을 권장해요. (이하 관련 내용 동일)

> By the way: Sometimes you need `this`
As mentioned above, you can often omit this, but there are times where it’s needed. If two functions have the same signature - that is, the same name, parameter types, and return type - then this can be helpful for choosing the right one. We’ll see more examples of this in the future.

ref. https://typealias.com/start/kotlin-receivers-and-extensions/

Reply:
BlackJackDealer에서 Deck에 접근해서 Deck의 데이터를 변경하고, 변경된 값을 가지도록 구현되어 있네요.
변경된 값을 내부에 가지고 있기 보다는 메시지를 보내서 확인하는 방법으로 변경해보면 어떨까요?

Reply:
> this는 주로 멤버변수 혹은 지역변수와 파라미터명이 동일할 때 사용하는 것을 권장해요. (이하 관련 내용 동일)

요 부분에 대해서는 개인적으로 this를 붙이는게 코드를 읽는사람에게 확신을 줄 수 있어서 좋다고 생각해요!
분명 생략해도 괜찮지만, 코드를 읽는 사람에게 생각할거리를 하나 줄여준다는 점에서 장점이 있는것 같습니다 🙌 
현석님께서는 이런 의견에 대해서 어떻게 생각하시나요 🤔

Reply:
저는 변수명이 겹치는 경우에만 사용하는 것이 좋다고 생각해요.
민영님이 말씀하신대로 의도는 좋다고 생각이 들지만, 전부 다 일일이 붙이는 것 또한 보일러 플레이트가 되기 때문에 저는 지양하는 것 같아요.
민영님이 작성하신 다른 함수들(예, BlackjackManager의 startGame함수)을 보시면 일부는 this가 붙어있고 일부는 this가 없어요.
```kotlin
    fun startGame(names: List<String>) {
        this.players = names.map { Player(name = it) }
        this.dealer = BlackjackDealer()
        dealer.shuffleCardDeck()
        dealer.startTheGame(players)
    }
```
이렇게 되면 오히려 코드를 읽는 사람에게 혼란을 주지 않을까요? 🤔